### PR TITLE
Add Kafka escapeHeaders documentation

### PR DESCRIPTION
**Pull request by antontroshin**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Add documentation for a new field for Kafka bindings and Pub/Sub

## Issue reference

https://github.com/dapr/components-contrib/issues/3503

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md [hunk 1]
```diff
@@ -99,6 +101,7 @@ spec:
 | `consumerFetchDefault` | N | Input/Output | The default number of message bytes to fetch from the broker in each request. Default is `"1048576"` bytes. | `"2097152"` |
 | `heartbeatInterval` | N | Input | The interval between heartbeats to the consumer coordinator. At most, the value should be set to a 1/3 of the `sessionTimeout` value. Defaults to `"3s"`. | `"5s"` |
 | `sessionTimeout` | N | Input | The timeout used to detect client failures when using Kafka’s group management facility. If the broker fails to receive any heartbeats from the consumer before the expiration of this session timeout, then the consumer is removed and initiates a rebalance. Defaults to `"10s"`. | `"20s"` |
+| `escapeHeaders` | N | Input | Enables URL escaping of the message header values received by the consumer. It allows receiving content with special characters that are usually not allowed in HTTP headers. Default is `false` | `true` |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/kafka.md]:
```suggestion
| `escapeHeaders` | N | Input | Enables URL escaping of the message header values received by the consumer. Allows receiving content with special characters that are usually not allowed in HTTP headers. Default is `false`. | `true` |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 2]
```diff
@@ -112,6 +114,7 @@ spec:
 | consumerFetchDefault | N | The default number of message bytes to fetch from the broker in each request. Default is `"1048576"` bytes. | `"2097152"` |
 | heartbeatInterval | N | The interval between heartbeats to the consumer coordinator. At most, the value should be set to a 1/3 of the `sessionTimeout` value. Defaults to "3s". | `"5s"` |
 | sessionTimeout | N | The timeout used to detect client failures when using Kafka’s group management facility. If the broker fails to receive any heartbeats from the consumer before the expiration of this session timeout, then the consumer is removed and initiates a rebalance. Defaults to "10s". | `"20s"` |
+| escapeHeaders | N | Enables URL escaping of the message header values received by the consumer. It allows receiving content with special characters that are usually not allowed in HTTP headers. Default is `false` | `true` |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
| escapeHeaders | N | Enables URL escaping of the message header values received by the consumer. Allows receiving content with special characters that are usually not allowed in HTTP headers. Default is `false`. | `true` |
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 3]
```diff
@@ -485,6 +488,36 @@ curl -X POST http://localhost:3500/v1.0/publish/myKafka/myTopic?metadata.correla
       }'
 ```
 
+## Receiving message headers with special characters
+
+The consumer application may be required to receive message headers that include special characters, which may cause HTTP protocol validation errors. 
+HTTP header values must follow specifications and some characters are not allowed, see [here](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
HTTP header values must follow specifications, making some characters not allowed. [Learn more about the protocols](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2).
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 4]
```diff
@@ -485,6 +488,36 @@ curl -X POST http://localhost:3500/v1.0/publish/myKafka/myTopic?metadata.correla
...
+In this case, you can enable `escapeHeaders` configuration setting which will use URL escaping to encode header values on the consumer side.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
In this case, you can enable `escapeHeaders` configuration setting, which uses URL escaping to encode header values on the consumer side.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 5]
```diff
@@ -485,6 +488,36 @@ curl -X POST http://localhost:3500/v1.0/publish/myKafka/myTopic?metadata.correla
...
+
+**Note**: When using this setting, the received message headers will be URL escaped, and you will need to URL unescape it to get the original value.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
When using this setting, the received message headers are URL escaped, and you need to URL "un-escape" it to get the original value.
{{% /alert %}}
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md [hunk 6]
```diff
@@ -485,6 +488,36 @@ curl -X POST http://localhost:3500/v1.0/publish/myKafka/myTopic?metadata.correla
...
+
+Setting `escapeHeaders` to `true`
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-apache-kafka.md]:
```suggestion
Set `escapeHeaders` to `true` to URL escape. 

```

**Review by hhunter-ms — DISMISSED** [review]
quick review

**Review by yaron2 — APPROVED** [review]

**Pull request merged**